### PR TITLE
Merge 2.1 into 3.1

### PR DESCRIPTION
**Pull request by jkotalik**
Another attempt at https://github.com/dotnet/aspnetcore/pull/18347 as I think the merge went wrong.

### CODE REVIEW

File: src/Security/CookiePolicy/test/CookieChunkingTests.cs [hunk 1]
```diff
@@ -44,6 +44,29 @@ public void AppendLargeCookie_WithOptions_Appended()
             Assert.Equal($"TestCookie={testString}; expires={now.AddMinutes(5).ToString("R")}; max-age=300; domain=foo.com; path=/bar; secure; samesite=strict; httponly", values[0]);
         }
 
+        [Fact]
+        public void AppendLargeCookie_WithOptions_Appended()
```
Comment (Tratcher) [inline src/Security/CookiePolicy/test/CookieChunkingTests.cs]: This is a duplicate test, remove it.

**Review by Tratcher — APPROVED** [review]

**Review by JunTaoLuo — APPROVED** [review]

File: src/SignalR/server/Core/src/HubConnectionContext.cs [hunk 2]
```diff
@@ -318,6 +318,7 @@ private ValueTask TryWritePingAsync()
                 return default;
             }
 
+            // TODO: cancel?
```
Comment (jkotalik) [inline src/SignalR/server/Core/src/HubConnectionContext.cs]: @BrennanConroy anything to be concerned about here?

**Comment by jkotalik** [comment]
Taking in favor of https://github.com/dotnet/aspnetcore/pull/18347, yeah?

File: src/SignalR/server/Core/src/HubConnectionContext.cs (hunk 2 above)
Comment (BrennanConroy) [reply src/SignalR/server/Core/src/HubConnectionContext.cs]: Haha, not really

**Review by BrennanConroy — APPROVED** [review]

**Comment by dougbu** [comment]
This is confusing. Why not force push the #18347 branch and keep all the comments in one place?

**Comment by jkotalik** [comment]
Because I wanted to make sure this PR was correct before squashing over @BrennanConroy . At this point, as it is already building, I'd rather just let it be and let it pass.

Also, many of the comments were talking about things that weren't relevant to the merge (things that existed in 3.1 that weren't changed).

**Comment by dougbu** [comment]
1. We're not going to squash anything -- this will be _merged_ 😈 
2. :ok: I'll review this one and ignore #18347
3. Please close #18347 as soon as you get relatively confident in the validation builds

**Review by dougbu — APPROVED** [review]

**Comment by dougbu** [comment]
Templates failures are discussed elsewhere. This is looking good so far.

**Comment by jkotalik** [comment]
So what should I do about the template failures? I'd like to get this in stat...

**Comment by BrennanConroy** [comment]
Looks like templates will be fixed once https://github.com/dotnet/aspnetcore/pull/18380 goes in

**Comment by BrennanConroy** [comment]
All but one of the templates tests passed on rerun: `BlazorServerTemplateWorks_IndividualAuth(useLocalDB: True)`

Looks unrelated to previous templates issues, is it a flaky test?

**Comment by rynowak** [comment]
I'm rerunning the template tests.

**Pull request merged**